Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Calibration/CalibrationDataHelper.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
+{
+    internal class CalibrationDataHelper
+    {
+        private const string ChessboardImageDirectoryName = "CHESS";
+        private const string ChessboardDetectedImageDirectoryName = "CHESS_DETECTED";
+        private const string DSLRArUcoDirectoryName = "DSLR_ARUCO";
+        private const string DSLRArUcoDetectedDirectoryName = "DSLR_ARUCO_DETECTED";
+        private const string HeadsetDataDirectory = "META";
+        private const string RootDirectoryName = "Calibration";
+
+        public static void Initialize(out int nextChessboardImageId, out int nextArUcoImageId)
```

Comment:
>nextChessboardImageId [](start = 46, length = 21)

Can you add some comments about the usage/intention of this next ID mechanism - AFAICT it looks like it's based on the number of files in a directory, which doesn't seem to be unique, so if this is truly an identifier I'm a bit confused about how we would prevent ID collisions.

Reply:
Ah, reading more it makes sense what it's doing (i.e. monotonically increasing ID based on the written out filenames).

---
In reply to: [281374506](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4189#discussion_r281374506) [](ancestors = 281374506)

Reply:
given this is an internal class I likely won't spend a great deal of time on updating this documentation. I don't think I will address this for now.